dashboard/app/models/user.rb

```diff
@@ -486,9 +486,8 @@ def email_and_hashed_email_must_be_unique
     # skip the db lookup if we are already invalid
     return unless errors.blank?
 
-    if ((email.present? && (other_user = User.find_by_email_or_hashed_email(email))) ||
-        (hashed_email.present? && (other_user = User.find_by_hashed_email(hashed_email)))) &&
-        other_user != self
+    if (email.present? && self != User.find_by_email_or_hashed_email(email)) ||
+      (hashed_email.present? && self != User.find_by_hashed_email(hashed_email))
```

Comment:
Note that this change introduces a subtle change in the logic, presumably for the better. Previously if both the email and the hashed email were present, we would ignore the `other_user` found by `email`. The new logic does not.